4.0.0-rc.6 AOT regression: exporting function named toString breaks AoT

**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
```

**Current behavior**

Exporting a function named `toString` [like so](https://github.com/ng-bootstrap/ng-bootstrap/blob/5856a75ace90c7323d073f1a9838560016bbb5d4/src/util/util.ts#L5) breaks AoT with the following error:

>   0% compilingCould not resolve module 'function toString() { [native code] }' relative to file /Users/pawelkozlowski/work/gitrepos/pkozlowski-opensource/core/src/util/util.ts
 10% building modules 3/3 modules 0 activeModuleNotFoundError: Module not found: Error: Can't resolve './../../$$_gendir/demo/src/app/app.module.ngfactory' in '/Users/pawelkozlowski/work/gitrepos/pkozlowski-opensource/core/demo/src'
    at factoryCallback (/Users/pawelkozlowski/work/gitrepos/pkozlowski-opensource/core/node_modules/webpack/lib/Compilation.js:230:39)


**Expected behavior**

Functions named `toString` are allowed or there is a precise documentation and clear AoT error when a "disallowed" function named is used.

**Minimal reproduction of the problem with instructions**

* clone https://github.com/pkozlowski-opensource/core/tree/ng_rc_6
* run `gulp build:demo` and see the error almost immediately.

This used to work in rc.5 and is linked to the function name (renaming the utility function in question from `toString` to sth like `stringify` makes the error go away.

**What is the motivation / use case for changing the behavior?**

Regression. 

**Please tell us about your environment:**

OSX, node v6.9.5

--- a/packages/compiler/src/aot/static_symbol_resolver.ts
+++ b/packages/compiler/src/aot/static_symbol_resolver.ts
@@ -224,7 +224,7 @@ export class StaticSymbolResolver {
           this.recordImportAs(symbol, importSymbol);
         }
 
-        const origin = origins[metadataKey];
+        const origin = origins.hasOwnProperty(metadataKey) && origins[metadataKey];
         if (origin) {
           // If the symbol is from a bundled index, use the declaration location of the
           // symbol so relative references (such as './my.html') will be calculated

--- a/packages/compiler/test/aot/compiler_spec.ts
+++ b/packages/compiler/test/aot/compiler_spec.ts
@@ -577,6 +577,7 @@ const FILES: MockData = {
       `,
       'app.module.ts': `
         import { NgModule }      from '@angular/core';
+        import { toString }      from './utils';
 
         import { AppComponent }  from './app.component';
 
@@ -585,6 +586,12 @@ const FILES: MockData = {
           bootstrap:    [ AppComponent ]
         })
         export class AppModule { }
+      `,
+      // #15420
+      'utils.ts': `
+        export function toString(value: any): string {
+          return  '';
+        }
       `
     }
   }

